Resources/Prototypes/explosion.yml

```diff
@@ -0,0 +1,41 @@
+#  Does not currently support prototype hot-reloading. See comments in c# file.
+
+- type: explosion
+  id: Default
+  damagePerIntensity:
+    types:
+      Heat: 5
+      Blunt: 5
+      Piercing: 5
+  tileBreakChance: [0, 0.5, 1]
+  tileBreakIntensity: [0, 10, 30]
```

Comment:
its a little unintuitive why this is an array or what the sequence of values corresponds to, imo

Reply:
So the way the tile break chance calculation works atm, it just a linearly interpolates between this list of points. So for this explosion type an intensity of 5 has a 25% chance to break a tile, 10 has 50%, 20 has a 75%, 30 has a 100%.
If a tile is broken, it then reduces the effective intensity and re-rolls the break chance, which allows it to break more than one tile at a time.

Making it a linearly interpolated list of points was the easiest way I could think of to allow people to somewhat intuitively customize tile break chances for different explosion types from yaml. Given that I'm not sure if this would even be used much I could also just hard code a function and add a toggle to disable tile breaking.

An alternative might be to just code in a distribution function whose mean or other properties scale with intensity, and then maybe allow someone to specify some of the properties of the distribution via yaml. This does mean the tile break chances are less customizable, but also much simpler I guess?

As an example, just taking the # of tiles to break by taking the floor of a number drawn from a symmetric triangular distribution with a lower cut-off at `intensity/30` and an overall scale of `1+intensity/50` leads to this destruction, which is comparable to the current default distribution:

![tile](https://user-images.githubusercontent.com/60421075/160223320-c084a3a5-75aa-4b1e-bef8-584126935ad4.png)

Reply:
I guess this also related to the comment about minibombs:
> minibombs are WAY too strong, they need their range reduced and to not space tiles at the very least. they should mostly be for precision bombing

Currently intensity is just directly tied to break chance. If someone wants a bomb that is capable of destroying walls, but also incapable of spacing tiles, they need to create a new explosion prototype and either tweak the damage values or tile break chance.

But for something like minibombs, it doesn't make much sense to define a new damage prototype, especially as that then involves storing more data in all the airtight maps and such. So I guess individual explosives also need some way of modifying the tile break chances?

All the tile break chance stuff is currently very arbitrary, any input / suggestions are appreciated